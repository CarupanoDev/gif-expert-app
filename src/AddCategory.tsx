// eslint-disable-next-line import/no-duplicates,no-use-before-define
import React from "react"
// eslint-disable-next-line import/no-duplicates
import { useState } from "react"

export const AddCategory = ({ setCategories }: any) => {

  const [inputValue, setInputValue] = useState<string>('');

  const handleInputChange = (e: React.FormEvent<HTMLInputElement>) => {
      setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2){
      setCategories(categories => [inputValue, ...categories])
      setInputValue('')
    }
  }

  return(

      <form onSubmit={handleSubmit} action="">
        <input
          type="text"
          value={ inputValue }
          onChange={handleInputChange}
        />
      </form>
  )
}
