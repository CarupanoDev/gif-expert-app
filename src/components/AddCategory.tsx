import { ChangeEvent, Dispatch, FormEvent, SetStateAction, useState } from "react"

export const AddCategory = (
  { setCategories }:
    {
      setCategories: Dispatch<SetStateAction<string[]>>
    }) => {

  const [inputValue, setInputValue] = useState<string>("")

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategories((categories: any) => [inputValue, ...categories])
      setInputValue("")
    }
  }

  return (

    <form onSubmit={handleSubmit} action="">
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}
