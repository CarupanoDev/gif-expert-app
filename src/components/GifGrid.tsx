import { GifGridItem } from "./GifGridItem"

import { useFetchGifs } from "../hooks/useFetchGifs"

interface GifGridProp {
  category: string
}

export const GifGrid = ({ category }: GifGridProp ) => {

  const { data: images, loading } = useFetchGifs( category )

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}

      <div className="card-grid">
          {
            images.map( image => (
              <GifGridItem
                // @ts-ignore
                key = {image.id}
                {...image}
              />
            ))
          }
      </div>
    </>
  )
}
