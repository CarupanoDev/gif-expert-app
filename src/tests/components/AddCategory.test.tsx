import { beforeEach, describe, expect, test, vi } from "vitest"
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe("Test  on AddCategory", () => {

  const setCategories = vi.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    vi.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test("it should show component correctly", () => {

    expect(wrapper).toMatchSnapshot()
  })

  test('it should change the input text', () => {
    const input = wrapper.find('input');
    const value = 'Hello, world!'

    input.simulate('change', {target: {value}});

    expect(wrapper.find('p').text().trim()).toBe(value)
  })

  test('No posted info on submit', () =>{
    wrapper.find('form').simulate('submit', {preventDefault(){}})

    expect(setCategories).not.toHaveBeenCalled();
  })

  test('Should call the setCategories methods and clean the input text', () => {
    const input = wrapper.find('input');
    const value = 'Hello, machine!'
    const value2 = ""

    input.simulate('change', {target: {value}});

    wrapper.find('form').simulate('submit', {preventDefault(){}})

    expect(setCategories).toHaveBeenCalled();

    expect(input.prop('value')).toBe(value2)
  })

})
