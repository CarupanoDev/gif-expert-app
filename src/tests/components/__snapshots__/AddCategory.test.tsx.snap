// Vitest Snapshot v1

exports[`Test  on AddCategory > it should show component correctly 1`] = `
<form
  action=""
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
